src/utils/xml.ts

```diff
@@ -1,29 +1,40 @@
 import { XMLParser } from "fast-xml-parser"
 
-/**
- * Parses an XML string into a JavaScript object
- * @param xmlString The XML string to parse
- * @returns Parsed JavaScript object representation of the XML
- * @throws Error if the XML is invalid or parsing fails
- */
+const XML_ENTITIES: Record<string, string> = {
+	"&": "&amp;",
+	"<": "&lt;",
+	">": "&gt;",
+	'"': "&quot;",
+	"'": "&apos;",
+}
+
+export function decodeSelectedXmlEntities(str: string, targets: string[] = ["&"]): string {
+	let result = str
+
+	for (const char of targets) {
+		const entity = XML_ENTITIES[char]
+		if (entity) {
+			result = result.replace(new RegExp(entity, "g"), char)
```

Comment:
```suggestion
			result = result.replaceAll(entity, char)
```

No need for regexes here.